### PR TITLE
fix(transaction): fix previous snapshot summary fetching

### LINKED ISSUE #2390: BUG: snapshot summary doesn't consider previous summary
### Apache Iceberg Rust version

976b689dd311c40986f72d98b95daccb5da37ef5

### Describe the bug

Hi team, IIRC, current snapshot summary only considers latest committed transaction and discard previous ones.
- Within a `SnapshotProducer`, `snapshot_id` is the id for snapshot to create, which gets initialized at producer [construction](https://github.com/apache/iceberg-rust/blob/976b689dd311c40986f72d98b95daccb5da37ef5/crates/iceberg/src/transaction/snapshot.rs#L133)
- When we try to get previous snapshot summary (to merge with current one), it's always return `None` [here](https://github.com/apache/iceberg-rust/blob/976b689dd311c40986f72d98b95daccb5da37ef5/crates/iceberg/src/transaction/snapshot.rs#L386-L387)

### To Reproduce

```rs
#[tokio::test]
    async fn test_transaction_snapshot_summary() {
        let catalog = new_memory_catalog().await;
        let table = make_v3_minimal_table_in_catalog(&catalog).await;

        let mut file_seq = 0u32;
        let mut append_file = |table: &crate::table::Table, record_count: u64, file_size: u64| {
            file_seq += 1;
            let file = DataFileBuilder::default()
                .content(DataContentType::Data)
                .file_path(format!("test/{file_seq}.parquet"))
                .file_format(DataFileFormat::Parquet)
                .file_size_in_bytes(file_size)
                .record_count(record_count)
                .partition(Struct::from_iter([Some(Literal::long(1))]))
                .partition_spec_id(0)
                .build()
                .unwrap();
            let tx = Transaction::new(table);
            tx.fast_append()
                .add_data_files(vec![file])
                .apply(tx)
                .unwrap()
        };

        let table = append_file(&table, /*record_count=*/ 10, /*file_size=*/ 100)
            .commit(&catalog)
            .await
            .unwrap();
        let table = append_file(&table, /*record_count=*/ 20, /*file_size=*/ 200)
            .commit(&catalog)
            .await
            .unwrap();

        let summary = &table
            .metadata()
            .current_snapshot()
            .unwrap()
            .summary()
            .additional_properties;

        assert_eq!(summary.get("total-records").unwrap(), "30");
        assert_eq!(summary.get("total-data-files").unwrap(), "2");
        assert_eq!(summary.get("total-files-size").unwrap(), "300");
    }
```
gets 
```sh
---- transaction::test::test_transaction_snapshot_summary stdout ----

thread 'transaction::test::test_transaction_snapshot_summary' (87414) panicked at crates/iceberg/src/transaction/mod.rs:636:9:
assertion `left == right` failed
  left: "20"
 right: "30"
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Expected behavior

_No response_

### Willingness to contribute

I can contribute a fix for this bug independently

### PR DESCRIPTION
**Pull request by dentiny**
## Which issue does this PR close?

- Closes https://github.com/apache/iceberg-rust/issues/2390

## What changes are included in this PR?

The root cause for the issue is, we use the current snapshot id (which hasn't been reflected to table metadata yet) to get previous snapshot summary.

## Are these changes tested?

Yes.

### CODE REVIEW

**Comment by dentiny** [comment]
The failure doesn't seem to be related to my change.
image

**Comment by timactive** [comment]
+1
 ```markdown — reproduced on a 484k-row table via fast_append. Without this
fix, `total-records` of the new snapshot was set to `added-records`
(=2 for a 2-row delta) instead of `parent.total + added` (=484065).
With the patch applied locally, the snapshot summary correctly
reports the cumulative and downstream consumers reading metadata.json
directly (iceberg-java in our case) see the right value on the next
commit. 36/36 snapshot tests stay green.

Edge case for the docs / future follow-up: tables with history
already corrupted by the buggy version will keep propagating wrong
totals because parent.total-records is read as-is. In practice
re-bootstrapping unblocks; a chain-walk recompute could self-heal
if needed, but it's not warranted for this PR. Thanks for the fix!```

**Comment by dentiny** [comment]
Hi @Xuanwo I'm wondering if you could take a look when you have some time? Thank you!

File: crates/iceberg/src/transaction/mod.rs [hunk 1]
```diff
@@ -584,3 +584,57 @@ mod test_row_lineage {
         assert_eq!(manifest_file.first_row_id, Some(30));
     }
 }
+
+#[cfg(test)]
+mod test {
```
Comment (CTTY) [inline crates/iceberg/src/transaction/mod.rs]: Can we merge this mod with `mod tests` above? https://github.com/apache/iceberg-rust/blob/main/crates/iceberg/src/transaction/mod.rs#L231

**Review by CTTY — CHANGES_REQUESTED** [review]
LGTM! just one comment

File: crates/iceberg/src/transaction/mod.rs (hunk 1 above)
Comment (dentiny) [reply crates/iceberg/src/transaction/mod.rs]: Yeah you're right, updated.

**Review by CTTY — APPROVED** [review]
LGTM! Thanks for the fix!
